src/osd/scheduler/mClockScheduler.cc

```diff
@@ -454,17 +468,63 @@ void mClockScheduler::enqueue(OpSchedulerItem&& item)
 
 void mClockScheduler::enqueue_front(OpSchedulerItem&& item)
 {
-  immediate.push_back(std::move(item));
+  unsigned priority = item.get_priority();
+  unsigned cutoff = get_io_prio_cut(cct);
+  auto id = get_scheduler_id(item);
+
+  if (priority >= cutoff || op_scheduler_class::immediate == id.class_id) {
+    immediate.push_back(std::move(item));
+  } else {
+    enqueue_strict(priority, std::move(item), true);
+  }
   // TODO: item may not be immediate, update mclock machinery to permit
   // putting the item back in the queue
 }
 
+void mClockScheduler::enqueue_strict(unsigned priority,
+                                     OpSchedulerItem&& item,
+				     bool front)
+{
```

Comment:
std::map<>::operator[] will default construct entries that don't exist, no need to check:

```
void mClockScheduler::enqueue_strict(unsigned priority,
                                     OpSchedulerItem&& item,
				     bool front)
{
  if (front) {
    high_priority[priority].push_front(std::move(item));
  } else {
    high_priority[priority].push_back(std::move(item));
  }
}
```

Reply:
Also, enqueue_strict places items into the high_priority queue -- let's call it enqueue_high.